### PR TITLE
Restore section and repeat block help in forms

### LINKED ISSUE #14061: Tool "section" and "repeat" element help text isn't displayed
**Describe the bug**
We have a section in our tool XML file defined as follows:
```
<section name="spins" expanded="true" title="Spins" help="Section for specifying a list of the spins to be used in the system ">
    <repeat name="spins" title="Species' spins" min="1" help="This should include a muon (mu) and one or more electrons (e)">
        <conditional name="spin_options">
            <param name="spin_preset" type="select" value="mu" label="Species" help="Select 'custom' to define own">
                <option selected="true" value="mu">mu</option>
                <option value="e">e</option>
                <option value="custom">custom</option>
            </param>
            <when value="custom">
                <param name="spin" optional="false" type="text" label="Species name"/>
                <param name="atomic_mass" optional="true" type="integer" min="0" value="" label="Atomic mass" help="Leave blank to use default mass - whole numbers only"/>
            </when>
            <when value="mu"/>
            <when value="e"/>
        </conditional>
    </repeat>
</section>
```
This is rendered like this:
image

Note that the `section`, `repeat` and `param` elements all have `help` text defined in the XML, but for the `section` and `repeat` elements this text isn't displayed anywhere.

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: dev
Commit: e394b1847e821a89b43bc6f15270dcedd67c5709

**Expected behavior**
Help text is displayed for sections and repeats.

### PR DESCRIPTION
**Pull request by guerler**
Fixes #14061. Ping @elichad, thank you for reporting this. Can you give this PR a try and see if it resolves the issue you observed. Thank you!

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

**Comment by elichad** [comment]
Thanks @guerler!

Section help looks good, but I noticed the repeat help is placed inside each repeat, when I had assumed it would appear only once (ideally between the repeat name and the Insert button or first repeat as indicated with arrows below).

I guess this is an ambiguous thing and either option can be helpful in different situations - has it been discussed before how the repeat help should be displayed?

image
image

**Comment by guerler** [comment]
@elichad I agree, it makes sense to just display it once under the title. I changed it. Can you try it out again? Thank you.

**Comment by elichad** [comment]
@guerler all looks good now, thanks! 

image

File: client/src/components/Form/FormDisplay.test.js [hunk 1]
```diff
@@ -105,4 +137,21 @@ describe("FormDisplay", () => {
         const conditionalBoolDisabled = wrapper.findAll("[type='checkbox']");
         expect(conditionalBoolDisabled.length).toEqual(0);
     });
+
+    it("repeats", async () => {
```
Comment (mvdbeek) [inline client/src/components/Form/FormDisplay.test.js:141]: `it` should form a sentence describing the test ;). Otherwise LGTM!

**Comment by mvdbeek** [comment]
We could, I'll leave that to @guerler who has a better idea if that'll work.

**Pull request merged**